src/db.ts

```diff
@@ -918,33 +917,40 @@ export abstract class Database {
           continue;
         }
 
-        updateJobs.recorded(values[1], values[0], values[4]);
-        batchedMs += 1;
-        batchedValues = batchedValues.concat(values);
-        if (batchedMs === Database.batchN) {
-          try {
-            const result = await this.recordMeasurementBatchedN(batchedValues);
-            recordedMeasurements += result;
-          } catch (err) {
-            // we may have concurrent inserts, or partially inserted data,
-            // where a request aborted
-            if (isUniqueViolationError(err)) {
-              recordedMeasurements += await this.recordMeasurementsFromBatch(
-                batchedValues
-              );
-            } else {
-              throw err;
-            }
+        let iMID = '';
+        // build a id for each unquie Measurement
+        iMID = iMID.concat(
+          run.id.toString() +
+            ' ' +
+            trial.id.toString() +
+            ' ' +
+            d.in.toString() +
+            ' ' +
+            criteria.get(m.c).id.toString()
+        );
+        if (iterationMap.has(iMID)) {
+          let value = iterationMap.get(iMID);
+          if (value !== undefined) {
```

Comment:
ehm, can this ever be `undefined`? Naively, by looking at your code, this should not happen, no?

Reply:
no it can never be undefined. this is a thing in typescrpit. its a try and catch i have to put in or it errors.

Reply:
Not sure what you mean with "try and catch", this isn't about exceptions, is it, it's simply about the return type of `.get()`, no?

If I understand correctly, you could just cast it to the type you know it is, cutting of the `| undefined` from the implicit type.